--- a/apps/DeepFaceLive/backend/CameraSource.py
+++ b/apps/DeepFaceLive/backend/CameraSource.py
@@ -123,10 +123,14 @@ class CameraSourceWorker(BackendWorker):
         if state.device_idx is not None and \
            state.driver is not None:
 
-            cv_api = {_DriverType.COMPATIBLE: cv2.CAP_ANY,
-                      _DriverType.DSHOW: cv2.CAP_DSHOW,
-                      _DriverType.MSMF: cv2.CAP_MSMF,
-                      _DriverType.GSTREAMER: cv2.CAP_GSTREAMER,
-                      }[state.driver]
+            if platform.system() == 'Darwin':
+                cv_api = cv2.CAP_ANY
+            else:
+                cv_api = {_DriverType.COMPATIBLE: cv2.CAP_ANY,
+                          _DriverType.DSHOW: cv2.CAP_DSHOW,
+                          _DriverType.MSMF: cv2.CAP_MSMF,
+                          _DriverType.GSTREAMER: cv2.CAP_GSTREAMER,
+                          }[state.driver]
 
             vcap = cv2.VideoCapture(state.device_idx, cv_api)
